### PR TITLE
Keep flash messages during (multiple) redirects.

### LINKED ISSUE #1162: Flash messages are not maintained with multiple redirects.
In continuation of [#1160](https://github.com/luckyframework/lucky/pull/1160#issuecomment-636313797).

### PR DESCRIPTION
**Pull request by wout**
## Purpose
Fixes #1162.

## Description
The `FlashStore#recycle` method from my proposal has been renamed to `FlashStore#keep`, but functionality is exactly the same. And I corrected a mistake I made with #1164 :blush:.

## Checklist
* [X] - An issue already exists detailing the issue/or feature request that this PR fixes
* [X] - All specs are formatted with `crystal tool format spec src`
* [X] - Inline documentation has been added and/or updated
* [X] - Lucky builds on docker with `./script/setup`
* [X] - All builds and specs pass on docker with `./script/test`

### CODE REVIEW

File: spec/lucky/action_redirect_spec.cr [hunk 1]
```diff
@@ -73,6 +73,16 @@ describe Lucky::Action do
...
   end
+
+  it "keeps flash messages for the next action" do
+    context = build_context_with_flash({success: "Keep me!"}.to_json)
+    action = RedirectAction.new(context, params)
+    response = action.redirect to: "/somewhere", status: 302
+    response.print
+
+    flash = Lucky::FlashStore.from_session(response.context.session)
+    flash.success.should eq("Keep me!")
```
Comment (paulcsmith) [inline spec/lucky/action_redirect_spec.cr:84]: Suggestion to change to match the other specs a bit more closely

```suggestion

    action = RedirectAction.new(context, params)
    response = action.redirect to: "/somewhere", status: 302
    response.print
    flash = Lucky::FlashStore.from_session(response.context.session)
    flash.success.should eq("Keep me!")
```

File: spec/lucky/cookies/flash_store_spec.cr [hunk 2]
```diff
@@ -47,6 +47,18 @@ describe Lucky::FlashStore do
     flash_store.success?.should be_nil
   end
 
+  describe "#keep" do
```
Comment (paulcsmith) [inline spec/lucky/cookies/flash_store_spec.cr:50]: I like this name 👍

File: spec/lucky/text_response_spec.cr [hunk 3]
```diff
@@ -16,32 +16,32 @@ describe Lucky::TextResponse do
...
-        print_response_with_body(context1)
+        print_response_with_body(context_1)
 
-        context2 = build_context
-        context2.session.set(Lucky::FlashStore::SESSION_KEY, next_json)
+        context_2 = build_context
+        context_2.session.set(Lucky::FlashStore::SESSION_KEY, next_json)
 
-        context2.flash.success.should eq("Yay!")
+        context_2.flash.success.should eq("Yay!")
```
Comment (paulcsmith) [inline spec/lucky/text_response_spec.cr:44]: Nice!

File: src/lucky/cookies/flash_store.cr [hunk 4]
```diff
@@ -22,6 +22,10 @@ class Lucky::FlashStore
     raise Lucky::InvalidFlashJSONError.new(session.get?(SESSION_KEY))
   end
 
+  def keep : Void
```
Comment (paulcsmith) [inline src/lucky/cookies/flash_store.cr:25]: We used `Void` in a few places, but typically that is reserved for doing C extensions and "idiomatic Crystal" is to use `Nil`. I don't think there is a technical difference, but 🤷  

```suggestion
  def keep : Nil
```

Comment (wout) [reply src/lucky/cookies/flash_store.cr:25]: Ok, good to know! I wasn't sure, but I saw it in some other places so I used it for consistency.

File: src/lucky/tags/base_tags.cr [hunk 5]
```diff
@@ -1,6 +1,6 @@
 module Lucky::BaseTags
   include Lucky::CheckTagContent
-  TAGS             = %i(a abbr address article aside b bdi body button code details dialog div dd dl dt em fieldset figcaption figure footer form h1 h2 h3 h4 h5 h6 head header html i iframe label li main mark menuitem meter nav ol option pre progress rp rt ruby s script section small span strong summary table template tbody td textarea th thead time title tr u ul video wbr)
+  TAGS             = %i(a abbr address article aside b bdi body button code details dialog div dd dl dt em fieldset figcaption figure footer form h1 h2 h3 h4 h5 h6 head header html i iframe label li main mark menuitem meter nav ol option pre progress rp rt ruby s script section small span strong summary table tbody td template textarea th thead time title tr u ul video wbr)
```
Comment (jwoertink) [inline src/lucky/tags/base_tags.cr:3]: 😂

Comment (wout) [reply src/lucky/tags/base_tags.cr:3]: 😁️

**Comment by paulcsmith** [comment]
Looks great!

**Pull request merged**